OpenRA.Mods.Common/Traits/Demolishable.cs

```diff
@@ -21,29 +22,47 @@ public class DemolishableInfo : IDemolishableInfo, ITraitInfo
 		[Desc("If true and this actor has EjectOnDeath, no actor will be spawned.")]
 		public readonly bool PreventsEjectOnDeath = false;
 
-		public object Create(ActorInitializer init) { return new Demolishable(init.Self, this); }
+		[GrantedConditionReference]
+		[Desc("Condition to grant during demolition countdown.")]
+		public readonly string Condition = null;
+
+		public object Create(ActorInitializer init) { return new Demolishable(this); }
 	}
 
-	public class Demolishable : IDemolishable, IPreventsEjectOnDeath
+	public class Demolishable : IDemolishable, INotifyCreated
 	{
 		readonly DemolishableInfo info;
+		ConditionManager conditionManager;
+		int conditionToken = ConditionManager.InvalidConditionToken;
 
-		public Demolishable(Actor self, DemolishableInfo info)
+		public Demolishable(DemolishableInfo info)
 		{
 			this.info = info;
 		}
 
-		public bool PreventsEjectOnDeath(Actor self)
+		void INotifyCreated.Created(Actor self)
+		{
+			conditionManager = self.TraitOrDefault<ConditionManager>();
+		}
+
+		void IDemolishable.Demolishing(Actor self, Actor saboteur)
 		{
-			return info.PreventsEjectOnDeath;
+			if (conditionManager != null && !string.IsNullOrEmpty(info.Condition))
```

Comment:
Previously the answer was no, because the lock prevented multiple demolitions from happening at once.  This isn't the case any more, though.  I think i'm going to have to do some fundamental work in this trait rather than pushing hacks around even more.